### DISCUSSION TITLE
Grid 2x2

### DISCUSSION THREAD

**Discussion by noahfraiture** [zellij-org/zellij#3645] [Q&A]
Hello, is it possible to have a different layout of pane without going through a layout file ? For instance I can't find how to do a 2x2. I can get 1 on left and 3 on right, but not 2x2. Is it something possible ?

**Comment by Lockszmith-GH** [comment]
<kbd>Ctrl</kbd>+<kbd>P</kbd> followed by <kbd>D</kbd> will create a pane below (down)
<kbd>Ctrl</kbd>+<kbd>P</kbd> followed by <kbd>R</kbd> will create a pane on the right

<kbd>Ctrl</kbd>+<kbd>P</kbd> followed by <kbd>D</kbd>
<kbd>Ctrl</kbd>+<kbd>P</kbd> followed by <kbd>R</kbd>
<kbd>Alt</kbd>+<kbd>Up Arrow</kbd>
<kbd>Ctrl</kbd>+<kbd>P</kbd> followed by <kbd>R</kbd>

Should get you a 2x2 grid.

**Reply by noahfraiture** [reply]
Oh great thank you. Is it possible to create in a stack ? Alt+[] works sometime but not always. I think it doesn't work with layouts

**Reply by Lockszmith-GH** [reply]
You'll need to define it as part of your layouts. I've never done it, but I'll take a look later.

**Comment by noahfraiture** [comment]
Hello, I find nothing in doc https://zellij.dev/documentation/keybindings-possible-actions or in the default config https://github.com/zellij-org/zellij/blob/main/zellij-utils/assets/config/default.kdl to create a new a pane stacked. Actually I found nothing about stacked pane except in layout.
The only thing that seem to have an impact is `Alt+ []` but it seems to have a very random behavior. It could be great to have a simple option to merge a pane with one next to it, to make a stack of pane.

**Reply by Lockszmith-GH** [reply]
Not sure where you got the term stack.
[stacked](https://zellij.dev/documentation/creating-a-layout.html?highlight=stack#stacked) panes are a specific term.

<kbd>Alt</kbd>+<kbd>[</kbd> or <kbd>]</kbd> in the config are designated:
```kdl
...
        bind "Alt [" { PreviousSwapLayout; }
        bind "Alt ]" { NextSwapLayout; }
...
```

There is a [whole section](https://zellij.dev/documentation/swap-layouts.html?swap-layouts) about it in the documentation site. See if this helps you achieve your goal.
If not, I'll be happy to read through it and experiment with you.

**Reply by noahfraiture** [reply]
Sorry for the wording. 
It doesn't seem to be a feature yet but I would like to be able to put the current pane in a stacked pane.
![image](https://github.com/user-attachments/assets/070e8615-4c19-4964-8f2f-350def6cc6e3)
![image](https://github.com/user-attachments/assets/2af991ae-16ab-40ee-b9d6-556544bbab97)
Like that. This is possible via `PreviousSwapLayout` `NextSwapLayout` but like it says, it's only layout.
It would be great to be able to do that regardless of layout configuration, but simply any time you want. I think [AeroSpace](https://nikitabobko.github.io/AeroSpace/guide) on macos works this way (I don't have it, I'm not sure).

You *merge* a pane, into another one to create stacked pane

**Reply by Lockszmith-GH** [reply]
As you've already noticed, stacking is possible, but you need to define a proper layout for it.

If I understand you correctly, you're looking for a dynamic stacking option.
That does seem like a new feature, maybe a plugin?

You might be able to 'roll your own' - by creating a bnunch of layouts and then rotating through them (with <kbd>Alt</kbd>+<kbd>[</kbd> or <kbd>]</kbd>)

**Reply by noahfraiture** [reply]
This is isn't the answer I was hoping for but thank you !
